#1148 include test coverage in actions

akaiap opened the pull request:
Capturing coveralls results to appear in the README.md as the badge.

dondi commented on .github/workflows/node.js.yml:
```diff
@@ -1,34 +1,58 @@
-# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
-# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
-
```
I’m curious about the removal of this comment—wouldn’t it be better to document what’s going on in this way? Or is it that the comment contains incorrect information?

akaiap replied:
Ah yes! It is correct-- I will add that back

dondi commented on test/grnstate-tests.js:
```diff
@@ -1,4 +1,5 @@
-import "chai/register-should";
+const chai = require("chai");
+chai.should(); // Activates "should" style assertions
```
Is there a reason that this is calling `require` rather than `import`? The latter is the newer approach. Also, we should move `chai.should()` to after all of the `import` statements because the convention is to do all imports first before executing code

akaiap replied:
- Since Mocha and the Node environment were still configured for CommonJS modules, using `require` made the test file run correctly under the regular CommonJS system. Switching to `import` would require the project to be recognized as an ES module (e.g., by setting "type": "module" in package.json).

- I will fix the import!

akaiap replied:
I tested the build again after moving the chai.should() call below all the import statements. While the build still ran, it triggered warnings and linting errors.

- The issue is from an inconsistent module system, where ES Modules (import) are being processed before CommonJS (require). Moving chai.should() below the imports affects the execution order in a way that conflicts with the current module setup.
- Linting failures are due to ESLint 9 deprecating old configuration formats:
ESLint now expects the "ignores" rule inside eslint.config.js instead of relying on the .eslintignore file.
The "languageOptions" object no longer allows "ecmaFeatures", requiring an updated configuration.

We can go down to V8, but may still encounter the same error.

dondi reviewed (commented):
Generally looks good, but with some questions and a small potential revision before we proceed. Let’s address these then do another round

dondi reviewed (commented):
Oh oof—I remember typing my comment last week during our meeting, but I failed to submit it…submitting now, sorry for the miss 😬

dondi commented:
Pinging on this—I think there are still some unresolved comments?

dondi reviewed (commented):
Deployment issues aside (as reported in #1148), the PR itself has been updated by running eslint’s automated `--fix` function, reducing existing linter errors down to 11, all of which say the same thing: `Parsing error: 'import' and 'export' may appear only with 'sourceType: module'`

Please look into this so that ideally we can “go green” on this PR then I think we will be ready to merge

akaiap commented:
Currently, so happy right now!! 
After months of getting the test coverage working, the badge is being updated, and coverage is passing!

Steps:
After cleaning up the code to now take imports/export of grnState, the code was pushed and passed all the checks! 

### Next steps: 
- While the checks pass -- I currently have ` continue-on-error: true # Linter failures won't stop the pipeline` in the yml package when Grnisight is built.  
- When I lint locally, I am receiving `1403 problems (1403 errors, 0 warnings)`
- Realized I was using two active and conflicting config files. _(.eslint.config.js and eslintrc.yml)_
- The next steps are to delete the old format (yml file) and reformat the config file to be able to use the `--fix command`

akaiap commented:
**ITS WORKING!** 

### Steps taken 
- Fixed module system config: Properly configured ESLint to support ES Modules w

- Resolved quote style issues: Fixed inconsistent quotes (single vs. double quotes)

- Proper configuration: Set up appropriate globals and rules in the eslint.config.js file 

Clean code base: the code now passes all linting checks, making it more maintainable and consistent :))))

dondi reviewed (approved):
LGTM
